addon/components/dropdown-link.hbs

```diff
@@ -1,13 +1,29 @@
-<LinkTo
-    @route={{@route}}
-    @models={{this.models}}
-    @activeClass={{this.activeClass}}
-    @query={{@query}}
-    @replace={{@replace}}
-    @current-when={{@currentWhen}}
-    class="dropdown-link dropdown-item"
-    {{on "click" this.onClick}}
-    ...attributes
->
-    {{yield}}
-</LinkTo>
\ No newline at end of file
+{{#if @model}}
+    <LinkTo
+        @route={{@route}}
+        @model={{@model}}
+        @activeClass={{this.activeClass}}
+        @query={{@query}}
+        @replace={{@replace}}
+        @current-when={{@currentWhen}}
+        class="dropdown-link dropdown-item"
+        ...attributes
+        {{on "click" this.onClick}}
+    >
+        {{yield}}
+    </LinkTo>
+{{else}}
+    <LinkTo
+        @route={{@route}}
+        @models={{this.models}}
+        @activeClass={{this.activeClass}}
+        @query={{@query}}
+        @replace={{@replace}}
+        @current-when={{@currentWhen}}
+        class="dropdown-link dropdown-item"
+        ...attributes
+        {{on "click" this.onClick}}
+    >
+        {{yield}}
+    </LinkTo>
+{{/if}}
```

Comment:
I think you could make this DRYer/cleaner by instead of the big if/else, do:
`@models={{if @model (array @model) @models}}`

Reply:
Or you might be able to just pass in both args separately and it’ll just work, as long as at least one of the values is undefined.

Reply:
@billdami I think it's more important to follow the pattern that LinkTo provides since we call out in the documentation that its essentially just a LinkTo component, and the LinkTo component takes in either `@model` or @models`.

I did try passing in both at first, but it complained so I had to switch it to what it is now

Reply:
Yeah I agree we should let the user pass in `@model` or `@models` to keep it consistent with the LinkTo component's api, but if internally we just do `@models={{if @model (array @model) @models}}` it should be functionally the same as what you have here w/o the duplicated code, and is just an internal implementation detail the user doesn't need to be aware of.

That being said i don't feel strongly enough about it to hold this up for that.